src/clr-angular/forms/datepicker/date-input.spec.ts

```diff
@@ -0,0 +1,288 @@
+                    expect(context.clarityElement.placeholder).toBe("MM/DD/YYYY");
+                });
+
+                it("binds the input type correctly", () => {
+                    expect(context.clarityElement.type).toBe("text");
+
+                    enabledService.fakeIsEnabled = false;
+                    context.detectChanges();
+
+                    expect(context.clarityElement.type).toBe("date");
+                });
+
+                it("listens to the input change events", () => {
+                    spyOn(context.clarityDirective, "onValueChange");
+
+                    const inputEl = context.fixture.debugElement.query(By.directive(ClrDateInput));
+                    inputEl.triggerEventHandler("change", inputEl);
+
+                    expect(context.clarityDirective.onValueChange).toHaveBeenCalled();
+                });
+            });
+        });
+
+        describe("Datepicker with ngModel", () => {
+            let fixture: ComponentFixture<TestComponentWithNgModel>;
+            let compiled: any;
+            let dateContainerDebugElement: DebugElement;
+            let dateInputDebugElement: DebugElement;
+
+            beforeEach(function() {
+                TestBed.configureTestingModule(
+                    {imports: [FormsModule, ClrFormsModule], declarations: [TestComponentWithNgModel]});
+
+                fixture = TestBed.createComponent(TestComponentWithNgModel);
+                fixture.detectChanges();
+                compiled = fixture.nativeElement;
+                dateContainerDebugElement = fixture.debugElement.query(By.directive(ClrDateContainer));
+                dateInputDebugElement = fixture.debugElement.query(By.directive(ClrDateInput));
+            });
+
+            it("accepts user input", fakeAsync(() => {
+                   const ioService: DateIOService = dateContainerDebugElement.injector.get(DateIOService);
+                   expect(ioService.date).toBeUndefined();
+
+                   fixture.componentInstance.dateValue = "01/02/2015";
+
+                   fixture.detectChanges();
+                   tick();
+
+                   dateInputDebugElement.nativeElement.dispatchEvent(new Event("change"));
+                   fixture.detectChanges();
+
+                   expect(ioService.date).not.toBeUndefined();
+                   expect(ioService.date.getFullYear()).toBe(2015);
+                   expect(ioService.date.getMonth()).toBe(0);
+                   expect(ioService.date.getDate()).toBe(2);
+
+                   fixture.componentInstance.dateValue = "01/02/201";
+
+                   fixture.detectChanges();
+                   tick();
+
+                   dateInputDebugElement.nativeElement.dispatchEvent(new Event("change"));
+                   fixture.detectChanges();
+
+                   expect(ioService.date).toBeNull();
+               }));
+
+            it("updates the input element value when the date is updated", () => {
```

Comment:
Testing that ngModelChange is emitted

Reply:
i believe this test already does that. please correct me if i am wrong.